src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlParameterCollection.cs

```diff
@@ -1,140 +0,0 @@
-// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MIT license.
-// See the LICENSE file in the project root for more information.
-
-using System;
-using System.Data;
-using System.Data.Common;
-
-namespace Microsoft.Data.SqlClient
-{
-    /// <include file='../../../../../../../doc/snippets/Microsoft.Data.SqlClient/SqlParameterCollection.xml' path='docs/members[@name="SqlParameterCollection"]/SqlParameterCollection/*' />
-    public sealed partial class SqlParameterCollection : DbParameterCollection
-    {
-        private bool _isDirty;
-        private static Type s_itemType = typeof(SqlParameter);
```

Comment:
Why remove this and instead call `typeof(SqlParameter` every time?

Reply:
The JIT can optimize away the `typeof(SqlParameter)` because at the time the method is compiled it knows the exactly value of the type handle and can insert it. If we put that handle into a variable then the jit only sees it as a variable access and will generate all the code needed to load the variable. It seems counter intuitive from a c# perspective to repeatedly use typeof rather than refer to a constant but it's actually faster.